compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt

```diff
@@ -386,7 +381,9 @@ internal class MetalRedrawer(
                 isForcedToPresentWithTransactionEveryFrame || interopTransaction.isNotEmpty()
             metalLayer.presentsWithTransaction = presentsWithTransaction
 
-            val mustEncodeAndPresentOnMainThread = presentsWithTransaction || waitUntilCompletion
+            // TODO: encoding on separate thread requires investigation for https://github.com/JetBrains/compose-multiplatform/issues/3862
```

Comment:
Let's create an issue in YouTrack for returning parallel rendering, and write a link to it here.

Otherwise we'll just forget about it.